### PR TITLE
osd/OSDMonitor: misc. fixes

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

**Comment by xiexingguo** [comment]
@tchaikov Can you please help to confirm this one too? Thanks!

File: src/mon/MonCommands.h [hunk 1]
```diff
@@ -628,7 +628,7 @@ COMMAND("osd erasure-code-profile ls", \
 	"list all erasure code profiles", \
 	"osd", "r", "cli,rest")
 COMMAND("osd set " \
-	"name=key,type=CephChoices,strings=full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|nodeep-scrub|notieragent|sortbitwise|require_jewel_osds", \
+	"name=key,type=CephChoices,strings=full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|nodeep-scrub|notieragent|sortbitwise|require_jewel_osds|require_kraken_osds", \
```
Comment (tchaikov) [inline src/mon/MonCommands.h]: probably you can reference this line: https://github.com/ceph/ceph/blob/9bef6cf/src/mon/OSDMonitor.cc#L6355, in the commit message.

i am not able to identify the piece of code referenced by https://github.com/ceph/ceph/pull/9248#issuecomment-221421724 anymore.

Comment (xiexingguo) [inline src/mon/MonCommands.h]: Sorry, I am trying to add a reference for why this change is necessary.
Will remove that later.

**Comment by tchaikov** [comment]
lgtm.

**Comment by xiexingguo** [comment]
retest this please

**Comment by yuriw** [comment]
re-run
http://pulpito.ceph.com/yuriw-2016-08-03_08:27:33-rados-wip-yuri-testing_2016_08_02-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2016-08-02_15:37:57-rados-wip-yuri-testing_2016_08_02-distro-basic-smithi/

**Pull request merged**